manifests/charts/istio-cni/templates/daemonset.yaml

```
@@ -41,6 +41,10 @@ spec:
         prometheus.io/scrape: 'true'
         prometheus.io/port: "15014"
         prometheus.io/path: '/metrics'
+        # Add AppArmor annotation
+        # This is required to avoid conflicts with AppArmor profiles
+        # which block certain privileged pod capabilities.
+        container.apparmor.security.beta.kubernetes.io/install-cni: unconfined
```

```suggestion
{{- if or (eq .Capabilities.KubeVersion.Version "") (and (le .Capabilities.KubeVersion.Major "1") (le .Capabilities.KubeVersion.Minor "31")) }}
        container.apparmor.security.beta.kubernetes.io/install-cni: unconfined
{{ end }}
```

If coupled with https://github.com/istio/istio/pull/53953/files#r1849057380 this will work in all scenarios and makes it clear what the K8S version constraints are, and will be forward compatible when K8S drops this label.

K8S version constraints around templating should be explicit in the template.

---

FWIW, I don't think kubernetes drops beta labels, but I'm not opposed to the gate regardless

---

The version is unreliable. Try running `helm template` with 3.16.2 -- it currently defaults to 1.31.

So anyone running `helm template` or `istioctl manifest generate` without  `--kube-version` (I would guess this represents >25% of users) is going to get the new behavior. Probably 95% of users are <1.31 right now, so ~23% of users will be broken if we do this

---

Ah, my version (`3.14`) does not default to anything but an empty string, and this works as intended. If they have changed this, nevermind then.